authzed/api/v1/core.proto

```diff
@@ -21,6 +22,21 @@ message Relationship {
 
   // subject is the subject to which the resource is related, in some manner.
   SubjectReference subject = 3 [ (validate.rules).message.required = true ];
+
+  // caveat is a reference to a the caveat that must be enforced over the tuple
+  ContextualizedCaveat caveat = 4 [ (validate.rules).message.required = false ];
+}
+
+/**
+ * ContextualizedCaveat represents a reference to a caveat used to by caveated tuples.
+ * The context are key-value pairs that will be injected at evaluation time.
+ */
+message ContextualizedCaveat {
+  /** caveat_id is the reference to the internal id of a stored caveat **/
+  uint64 caveat_id = 1 [(validate.rules).uint64.gt = 0];
```

Comment:
This needs to be the caveat's name, since this is the API that will be used publicly.